core/directory.go

```diff
@@ -282,27 +262,33 @@ func (dir *Directory) Entries(ctx context.Context, src string) ([]string, error)
 		return nil, fmt.Errorf("%s: no such file or directory", src)
 	}
 
-	entries, err := ref.ReadDir(ctx, bkgw.ReadDirRequest{
-		Path: src,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	useSlash, err := SupportsDirSlash(ctx)
-	if err != nil {
-		return nil, err
+	bkSessionGroup, ok := buildkit.CurrentBuildkitSessionGroup(ctx)
+	if !ok {
+		return nil, fmt.Errorf("no buildkit session group in context")
 	}
 
 	paths := []string{}
-	for _, entry := range entries {
-		path := entry.Path
-		if useSlash && os.FileMode(entry.Mode).IsDir() {
-			path += "/"
+	err = MountRef(ctx, ref, bkSessionGroup, func(root string) error {
```

Comment:
Little suggestion - I'm noticing as we convert more, there's some instances of repetition. If you have a moment, if you have any good ideas for isolating the "logic" of the operations, that would be very neat.

Helpers like `MountRefInCurrentSession` for example.